examples/vite-react-avatar-stack/src/components/Avatars.tsx

```diff
@@ -0,0 +1,217 @@
+            ? user.profileData.memberColor
+            : "rgb(229 231 235)",
+        }}
+        className={styles.userInfoContainer}
+        id="avatar"
+      >
+        <p
+          style={{ color: user.isConnected ? "#fff" : "rgb(156 163 175)" }}
+          className={styles.smallText}
+        >
+          {initials}
+        </p>
+      </div>
+
+      {/* 💡 Display the name of the user from the `profileData` object 💡 */}
+      <div id="user-list" className={styles.userList}>
+        <p className={styles.name}>{name}</p>
+        <div className={styles.wrapper}>
+          <div
+            style={{
+              backgroundColor: user.isConnected
+                ? "rgb(34 197 94)"
+                : "rgb(100 116 139)",
+            }}
+            className={styles.statusIndicator}
+          />
+          <p className={styles.statusIndicatorText}>{statusIndicatorText}</p>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const Surplus: FunctionComponent<{ otherUsers: Member[] }> = ({
+  otherUsers,
+}) => {
+  const [showList, setShowList] = useState(false);
+  const { listRef, plusButtonRef } = useClickOutsideList(() =>
+    setShowList(false),
+  );
+
+  return otherUsers.length > MAX_USERS_BEFORE_LIST ? (
+    <div className={styles.surplusContainer}>
+      <div
+        className={styles.badge}
+        style={{
+          zIndex: otherUsers.length + 50,
+        }}
+        ref={plusButtonRef}
+        onClick={() => {
+          setShowList(!showList);
+        }}
+      >
+        +{otherUsers.slice(MAX_USERS_BEFORE_LIST).length}
+      </div>
+
+      {showList ? (
+        <div className={styles.list} ref={listRef}>
+          {otherUsers.slice(MAX_USERS_BEFORE_LIST).map((user) => (
+            <div className={styles.user} key={user.clientId}>
+              <UserInfo user={user} />
+            </div>
+          ))}
+        </div>
+      ) : null}
+    </div>
+  ) : null;
+};
+
+const Avatars = ({
+  otherUsers,
+  self,
+}: {
+  otherUsers: Member[];
+  self: Member | null;
+}) => {
+  const [hover, setHover] = useState(false);
+  const totalWidth = calculateTotalWidth({ users: otherUsers });
+
+  return (
+    <div className={styles.avatarsContainer} style={{ width: `${totalWidth}px` }}>
+      <div
+        className={styles.avatar}
+        onMouseOver={() => setHover(true)}
+        onMouseLeave={() => setHover(false)}
+      >
+        <p className={styles.name}>You</p>
+        <div className={styles.statusIndicatorOnline} id="status-indicator" />
+
+        {hover && self ? (
+          <div className={styles.popup}>
+            <UserInfo user={self} isSelf={true} />
+          </div>
+        ) : null}
+      </div>
+      <OtherAvatars
+        usersCount={otherUsers.length}
+        users={otherUsers.slice(0, MAX_USERS_BEFORE_LIST).reverse()}
+      />
+      {/** 💡 Dropdown list of surplus users 💡 */}
+      <Surplus otherUsers={otherUsers} />
+    </div>
+  );
+};
+
+export default Avatars;
```

Comment:
I understand we want less files, and I am on board with this, but now we have very long files that are harder to understand. Given these examples are meant to be read and understood by devs, I am concerned this is too complicated.